esx_service/vmdk_ops.py

```diff
@@ -153,12 +157,24 @@ def RunCommand(cmd):
 
     return (s, o)
 
+def is_valid_volume_name(name):
+    """ Check the given name is a valid volume name """
+    pattern = "^" + VALID_VOLUME_NAME_REG + "$"
+    valid_name_reg = re.compile(pattern)
+    if valid_name_reg.match(name):
+        return True
+    else:
+        return False
 
 # returns error, or None for OK
 # opts is  dictionary of {option: value}.
 # for now we care about size and (maybe) policy
 def createVMDK(vmdk_path, vm_name, vol_name, opts={}, vm_uuid=None, tenant_uuid=None, datastore=None):
     logging.info("*** createVMDK: %s opts = %s", vmdk_path, opts)
+    if not is_valid_volume_name(vol_name):
```

Comment:
parse_vol_name does more that that - it also separated DS name and handles them differnetly.
And no, I see no reason to mimic local local driver, we have deviated from it already by adding `@` to full volume name.